aa-sdk/core/src/middleware/erc7677middleware.ts

```diff
@@ -159,21 +185,143 @@ export function erc7677Middleware<
 
   const paymasterAndData: ClientMiddlewareFn = async (
     uo,
-    { client, account, feeOptions, overrides }
+    { client, account, feeOptions, overrides, policyToken }
   ) => {
     const userOp = deepHexlify(await resolveProperties(uo));
-    const context =
+
+    let context =
       (typeof params?.context === "function"
         ? await params?.context(userOp, { overrides, feeOptions })
         : params?.context) ?? {};
+    const entrypoint = account.getEntryPoint();
 
     if (!client.chain) {
       throw new ChainNotFoundError();
     }
 
+    let max_token: BigInt | undefined = undefined;
+    if (policyToken !== undefined) {
+      let total_gas = BigInt(0);
+      total_gas += uo.callGasLimit
+        ? BigInt(uo.callGasLimit.toLocaleString())
+        : BigInt(0);
+
+      total_gas += uo.preVerificationGas
+        ? BigInt(uo.preVerificationGas.toLocaleString())
+        : BigInt(0);
+
+      total_gas += uo.verificationGasLimit
+        ? BigInt(uo.verificationGasLimit.toLocaleString())
+        : BigInt(0);
+
+      let paymasterAddress = "0x";
```

Comment:
(Nit) If this variable is always going to be assigned to later, then don't initialize it with an assignment. Instead do
```ts
let paymasterAddress: Hex;
```
This has two advantages:
* It gives it the more narrow type `Hex` instead of `string`, meaning it would be a type error if we accidentally assign a non-hex value to it later.
* It makes it a type error if we forget to assign to it in one of the branches below.